src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.pip.subscription.management.controllers;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.pip.subscription.management.models.Subscription;
+import uk.gov.hmcts.reform.pip.subscription.management.services.SubscriptionService;
+
+import java.util.List;
+
+
+@RestController
+@Api(tags = "Subscription management API")
+@RequestMapping("/subscription")
+public class SubscriptionController {
+
+    @Autowired
+    SubscriptionService subscriptionService;
+
+    @PostMapping(consumes = "application/json", produces = "application/json")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "Subscription created"),
+    })
+    public ResponseEntity<Subscription> createSubscription(@RequestBody Subscription sub) {
+        /*
+          generate new unique subscription with json - 'id' field is hidden as auto-generated
+         */
```

Comment:
javadocs sit above the method, not inside.
but with the controllers we dont really need the javadocs in as we use swagger annotations, if you add the annotation @ApiOperation you can add in what that api is supposed to do and then that will be included on the swagger page too, although that description might need changing up a bit as its not too clear on what the purpose is and the hidden id field

Reply:
Replaced all these javadocs with @ApiOperations and hopefully made them a bit more descriptive.